context.go

```diff
@@ -34,6 +34,21 @@ func NewRouteContext() *Context {
 	return &Context{}
 }
 
+// WithRouteContext returns the list of methods allowed for the current
+// request, based on the current routing context.
+func AllowedMethods(ctx context.Context) []string {
+	if rctx := RouteContext(ctx); rctx != nil {
+		result := make([]string, 0, len(rctx.methodsAllowed))
+		for _, method := range rctx.methodsAllowed {
+			if method := methodTypString(method); method != "" {
+				result = append(result, method)
+			}
+		}
```

Comment:
Hi, do you think it'd make sense to expose a method on `rctx` instead?

```go
rctx := chi.RouteContext(ctx)

var methods []string = rctx.AllowedMethods()
```

Reply:
That would serve my purpose just as well. Would you like me to update the PR?

Reply:
sounds good :) thank you

Reply:
I've made the requested change, and rebased on a current version of the `master` branch.